packages/klighd-core/src/search-bar/searchbar-panel.tsx

```diff
@@ -0,0 +1,707 @@
+                                this.tooltipEl.style.display = 'none'
+                            }
+                        },
+                        click: () => {
+                            this.panToElement(result.id)
+                            this.selectedIndex = index
+                            this.update()
+                        }
+                    },
+                    hook: {
+                        insert: vnode => {
+                            vnode.elm!.addEventListener('mouseover', () => {
+                                (vnode.elm as HTMLElement).style.backgroundColor = 'rgba(0, 0, 0, 0.25)'
+                            })
+                            vnode.elm!.addEventListener('mouseout', () => {
+                                (vnode.elm as HTMLElement).style.backgroundColor = 'transparent'
+                            })
+
+                            if (isSelected) {
+                                (vnode.elm as HTMLElement).scrollIntoView({
+                                    behavior: 'smooth',
+                                    block: 'nearest'
+                                })
+                            }
+                        },
+                        update: (oldVnode, vnode) => {
+                            if (isSelected) {
+                                setTimeout(() => {
+                                    (vnode.elm as HTMLElement).scrollIntoView({
+                                        behavior: 'smooth',
+                                        block: 'nearest'
+                                    })
+                                }, 0)
+                            }
+                        }
+                    }
+                }, [this.textRes[index],
+                    isSelected ? h('div', {
+                        style: {
+                            fontSize: '11px',
+                            color: '#007acc',
+                            marginTop: '4px'
+                        }
+                    }, [this.decodeId(result.id)]) : null
+                ])
+            }))
+        ])
+    }   
+
+    /**
+     * Zoom in on a certain element.
+     * @param elementId id of the element to zoom in to. 
+     */
+    private panToElement(elementId: string): void {
+        const action: FitToScreenAction = {
+            kind: 'fit',
+            elementIds: [elementId],
+            animate: true,
+            padding: 20,
+            maxZoom: 2
+        }
+        this.actionDispatcher.dispatch(action)
+    }
+
+    /** Resets the UI by removing tooltips and the result list */
+    private resetUI(): void {
+        const input = document.getElementById('search') as HTMLInputElement | null
+        const tagInput = document.getElementById('tag-search') as HTMLInputElement | null
+
+        if (input) input.value = ''
+        if (tagInput) tagInput.value = ''
+        if (this.tooltipEl) this.tooltipEl.style.display = 'none'
```

Comment:
like above, you can store these values in this class and update the content during the rendering loop, that way you never need to use the `document` search